staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -37,16 +37,20 @@ import (
 	"k8s.io/apiserver/pkg/authorization/authorizer"
 )
 
+const (
+	maxUserAgentLength      = 1024
```

Comment:
Can we put this at 8k so it's in line with other defaults ie apache:
http://httpd.apache.org/docs/2.2/mod/core.html#limitrequestfieldsize

I realize most use cases may only need 1k, however I specifically started this issue (https://github.com/kubernetes/kubernetes/issues/64791#issue-329640478) so that I could send a very specific per call user-agent.

:pray: /cc @hzxuzhonghu @tallclair